python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -967,10 +970,16 @@ def _setFlushed(self):
         """Un-Set the dirty flag all of the way down the schema."""
         if self._schema:
             for k in self._schema.allItemNames():
+                ## Avoid attributes the likes of job.master which crawl back up the tree
+                if self._schema[k].hasProperty('visitable'):
+                    if (not self._schema[k].getProperties()['visitable'] or self._schema[k].getProperties()['visitable'] == 0):
+                        continue
+                if self._schema[k].hasProperty('transient'):
+                    if(self._schema[k].getProperties()['transient'] or self._schema[k].getProperties()['transient'] == 1):
```

Comment:
Yeah, that's annoying but it's a also a good reason to avoid `is True` when doing boolean checks and instead rely on Python's duck typing most of the time. http://rhodesmill.org/brandon/slides/2012-11-pyconca/ makes interesting reading on this part.